R/genbank2uid.R

```diff
@@ -39,61 +34,78 @@ genbank2uid <- function(id, batch_size = 100, ...) {
   process_batch <- function(id, ...) {
     #removes version number of accession ids
     id <- gsub(pattern = "\\.[0-9]+$", "", id)
-    url2 <- paste0(
+    
+    # Make NCBI eutils query
+    query <- paste0(
       ncbi_base(),
-      "/entrez/eutils/elink.fcgi?dbfrom=nucleotide&db=taxonomy&id=")
-    query <- paste0(url2, paste(id, collapse = "&id="))
-    res <- GET(query, ...)
-    stop_for_status(res)
-    result <- xml_text(xml_find_all(read_xml(con_utf8(res)),
-                                    "//LinkSetDb//Link//Id"))
-    if (length(result) > length(id)) {
-      url3 <- paste0(
-        ncbi_base(),
-        "/entrez/eutils/esummary.fcgi?db=taxonomy&id=")
-      query <- paste0(url3, paste(result, collapse = ","))
-      res <- GET(query, ...)
-      stop_for_status(res)
-      nn <- xml_find_all(read_xml(con_utf8(res)), "//eSummaryResult//DocSum")
-      result <- lapply(nn, function(z) {
-        list(
-          name = xml_text(xml_find_all(z, "Item[@Name=\"ScientificName\"]")),
-          id = xml_text(xml_find_all(z, "Item[@Name=\"TaxId\"]"))
-        )
-      })
-    }
-    if (length(result) < length(id)) {
-      result <- rep(NA_character_, length(id))
+      "/entrez/eutils/esummary.fcgi?db=nucleotide&id=",
+      paste(id, collapse = ",")
+    )
+    
+    # Execute query
+    xml_result <- GET(query, ...)
+    stop_for_status(xml_result)
+    parsed_xml <- read_xml(con_utf8(xml_result))
+    
+    # Extract taxon ID and sequences name
+    taxon_ids <- xml_text(xml_find_all(parsed_xml,
+                                       "//eSummaryResult//DocSum//Item[@Name='TaxId']"))
+    titles <- xml_text(xml_find_all(parsed_xml,
+                                    "//eSummaryResult//DocSum//Item[@Name='Title']"))
+    
+    # Add NAs for failed queries
+    raw_errors <- xml_text(xml_find_all(parsed_xml, "//eSummaryResult//ERROR"))
+    if (length(raw_errors) > 0) {
+      error_regexes <- c('^Invalid uid (.+) at position=[0-9]+$', 
```

Comment:
We can add regexes to this if more error types pop up.

Hmm, I dont know how to use this review tool right. The variable `error_regexes` was what I was talking about.